numpy/lib/function_base.py

```diff
@@ -2300,7 +2320,15 @@ def __init__(self, pyfunc, otypes=None, doc=None, excluded=None,
         else:
             self._in_and_out_core_dims = None
 
-    def __call__(self, *args, **kwargs):
+    def _init_stage_2(self, pyfunc, *args, **kwargs):
+        self.__name__ = pyfunc.__name__
+        self.pyfunc = pyfunc
+        if self._doc is None:
+            self.__doc__ = pyfunc.__doc__
+        else:
+            self.__doc__ = self._doc
```

Comment:
Using ```functools.wraps``` isn't actually necessary here, since we're making a decorator for a class instead of a function. That being said, your comment made me realize that I didn't have a test that checked to make sure the docstring is properly preserved after using @vectorize.

Reply:
We have a callable object, so that is basically equivalent to a function.  An argument may be that `__wrapped__` is not strictly true and e.g. inheriting the signature annotations might actually be wrong.  So in that sense it may be better to maybe use it, but pick out the interesting attributes for now, which the `functools.wraps` doc list (or `updated_wrapper`).

When used as a decorator, `__module__` is a very interesting attribute because it is a step towards allowing pickling, and if we use name, qualname is also.